sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineMockTests.java

```diff
@@ -0,0 +1,195 @@
+        WireMockServer mockServer = startMockServer();
+
+        try {
+            ComputeManager computeManager = mockComputeManager(mockServer);
+
+            PagedIterable<VirtualMachine> virtualMachines = computeManager.virtualMachines().listByVirtualMachineScaleSetId("/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/javacsmrg97796/providers/Microsoft.Compute/virtualMachineScaleSets/vmss035803b7");
+            // 1 element per page, 2 pages in total
+            Assertions.assertEquals(2, virtualMachines.stream().count());
+            Assertions.assertTrue(stateHolder.firstPageRequested);
+            Assertions.assertTrue(stateHolder.secondPageRequested);
+        } finally {
+            if (mockServer.isRunning()) {
+                mockServer.shutdown();
+            }
+        }
+    }
+
+    private ComputeManager mockComputeManager(WireMockServer mockServer) {
+        Map<String, String> environment = new HashMap<>();
+        environment.put("resourceManagerEndpointUrl", mockServer.baseUrl());
+        // just to mitigate npe, no actual use here
+        environment.put("microsoftGraphResourceId", mockServer.baseUrl());
+        AzureProfile mockProfile = new AzureProfile(UUID.randomUUID().toString(), UUID.randomUUID().toString(), new AzureEnvironment(environment));
+        ComputeManager computeManager = ComputeManager.authenticate(new HttpPipelineBuilder().build(), mockProfile);
+        return computeManager;
+    }
+
+    private WireMockServer startMockServer() {
+        ResponseTransformer transformer = new ResponseTransformer() {
+            @Override
+            public Response transform(Request request, Response response, FileSource fileSource, Parameters parameters) {
+                Map<String, Object> responseBody = new HashMap<>();
+                Map<String, Object> vm;
+                VirtualMachineInner vmInner = mockVmInner();
+                try {
+                    vm = SERIALIZER.deserialize(SERIALIZER.serialize(vmInner, SerializerEncoding.JSON), Map.class, SerializerEncoding.JSON);
+                } catch (IOException e) {
+                    return failedResponse(e.getMessage());
+                }
+                vm.put("name", "vmName");
+                if (request.getUrl().contains("Microsoft.Compute/virtualMachines")) {
+                    // first page
+                    stateHolder.firstPageRequested = true;
+                    responseBody.put("value", Collections.singletonList(vm));
+                    responseBody.put("nextLink", stateHolder.nextLinkUrl);
+                    return successResponse(responseBody);
+                } else if (request.getUrl().contains(NEXT_LINK_PATH)) {
+                    // next link page
+                    stateHolder.secondPageRequested = true;
+                    // check the nextLink is actually encoded
+                    if (ResourceUtils.encodeResourceId(request.getUrl()).equals(request.getUrl())) {
```

Comment:
Since you already use encodeResourceId to process in code, should you use another way to verify it is valid URL? `new URI` e.g.?

Reply:
Sure, will do `new URI` test here.